routes/web.php

```diff
@@ -11,10 +11,24 @@
 |
 */
 
+use Illuminate\Routing\Router;
+
 Route::get('/', function () {
     return view('welcome');
 });
 
 Auth::routes();
 
 Route::get('/home', 'HomeController@index')->name('home');
+
+$router->group([
+    'middleware' => ['auth'],
+    'prefix' => 'api',
+    'as' => 'api.'
+], function (Router $router) {
+    $router->get('/transactions/period/{period1}/{period2}', 'APIController@periodToPeriod')->name('periodToPeriod');
+    $router->get('/transactions/recent', 'APIController@recentTransactions')->name('recentTransactions');
```

Comment:
i feel like `recent` is a bit ambiguous in terms of time scale? a replacement could be 'pastMonth` maybe?

Reply:
I personally think it's okay, perhaps discussing with the client as to their preferred data range. I provided this method as on first load we wouldn't only want to return the "recent" data rather than the whole dataset on first load of the dashboard.

Reply:
good idea, and to be fair this api would be used in conjunction with a label which mentioned the time frame somewhere on the UI

Reply:
Yeah, we'd likely have like a date selector the client would use to chose their own date ranges and then we'd likely just validate it pass it to this api and return it